sdk/src/assertions/multi_asset_hash.rs

```diff
@@ -0,0 +1,360 @@
+                .get_assertion_from_link(&part.hash_assertion.url())
+                .ok_or_else(|| {
+                    Error::C2PAValidation(ASSERTION_MULTI_ASSET_HASH_MISSING_PART.to_string())
+                })?;
+
+            let label = assertion.label();
+
+            match &part.location {
+                LocatorMap::ByteRangeLocator(locator) => {
+                    let offset = locator.byte_offset;
+                    let length = locator.length;
+
+                    // Read only the specified parts within the larger stream.
+                    reader.seek(std::io::SeekFrom::Start(offset))?;
+                    let buf = reader.read_to_vec(length).map_err(|_| {
+                        Error::C2PAValidation(ASSERTION_MULTI_ASSET_HASH_MISSING_PART.to_string())
+                    })?;
+                    let mut part_reader = Cursor::new(buf);
+
+                    // Perform validation on each part depending on type of hash.
+                    match label.as_str() {
+                        l if l.starts_with(DataHash::LABEL) => {
+                            let dh = DataHash::from_assertion(assertion)?;
+                            let alg = match &dh.alg {
+                                Some(alg) => alg,
+                                None => claim.alg(),
+                            };
+                            dh.verify_stream_hash(&mut part_reader, Some(alg))?;
+                        }
+                        l if l.starts_with(BoxHash::LABEL) => {
+                            let bh = BoxHash::from_assertion(assertion)?;
+                            let box_hash_processor = asset_handler
+                                .ok_or(Error::UnsupportedType)?
+                                .asset_box_hash_ref()
+                                .ok_or(Error::HashMismatch("Box hash not supported".to_string()))?;
+                            bh.verify_stream_hash(
+                                &mut part_reader,
+                                Some(claim.alg()),
+                                box_hash_processor,
+                            )?;
+                        }
+                        l if l.starts_with(BmffHash::LABEL) => {
+                            return Err(Error::NotImplemented(
+                                "BmffHash not yet implemented for Multi-Asset hashes".to_string(),
+                            ));
+                        }
+                        _ => {}
+                    }
+                }
+                LocatorMap::BmffBox { .. } => {
+                    return Err(Error::NotImplemented(
+                        "BmffBox locators not yet implemented for Multi-Asset hashes".to_string(),
+                    ));
+                }
+            }
+        }
+
+        Ok(())
+    }
+}
+
+#[derive(Deserialize, Serialize, Debug, PartialEq, Eq)]
+pub struct PartHashMap {
+    pub location: LocatorMap,
+    #[serde(rename = "hashAssertion")]
```

Comment:
Not a blocker, but I think it makes more sense to have `#[serde(rename_all = "camelCase")]` on the whole struct. Same goes for the other places this pattern is used.